Update VideoTexture doc with note about using RGBA format to avoid Firefox performance issue.

jlivak opened the pull request:
There's an issue with Firefox desktop that causes RGBFormat on a video texture to destroy performance.  When testing with a single 4K 360 video, frame times on Firefox with RGBFormat were ~200x RGBAFormat.  (62ms vs .32ms)  

I propose defaulting to RGBAFormat to avoid this.

jlivak committed:
VideoTexture should use RGBA format to avoid firefox issue.

There's an issue with Firefox desktop that causes RGBFormat on a video texture to destroy performance.  When testing with a single 4K 360 video, frame times on Firefox with RGBFormat were ~200x RGBAFormat.  (62ms vs .32ms)  I propose defaulting to RGBAFormat to avoid this.

Mugen87 commented:
How about reporting this performance issue to Mozilla instead?

https://bugzilla.mozilla.org/

As long as no other browser are affected, it's better to keep the more correct default value `RGBFormat`. Simply because most videos do not need an alpha channel.

jlivak commented:
There definitely should be a mozilla bug report as well, however the main concern here is just helping people who want a quick video texture implementation to have it working across browsers without needing to hunt down solutions for per-browser issues.

If you'd prefer to keep the more sensible value of RGBFormat, even if it means VideoTexture doesn't work across-browser out of the box, maybe just a callout in the docs page for VideoTexture that the filter defaults to RGB but if performance issues are encountered to try changing it to RGBA?

Mugen87 commented:
Adding a note to the docs sounds good! The `format` parameter is listed in the ctor description. I think this would be a good place to mentioned the default `RGBFormat` and this issue.

jlivak committed:
Revert VideoTexture.js changes.

jlivak committed:
Update VideoTexture doc with 'format' default value and note about a Firefox bug.

jlivak commented:
Alright cool, that works for me.  Reverted the VideoTexture.js change and updated the VideoTexture doc page.

Also mozilla bug report for future visibility:  https://bugzilla.mozilla.org/show_bug.cgi?id=1708491

aardgoose commented:
I suspect this is a similar issue to this one:

[https://bugzilla.mozilla.org/show_bug.cgi?id=1486454](https://bugzilla.mozilla.org/show_bug.cgi?id=1486454)

mrdoob commented:
Thanks!
